bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/FileInfo.java

```diff
@@ -112,6 +114,29 @@ public long getSizeSinceLastwrite() {
         return sizeSinceLastwrite;
     }
 
+    public ByteBuffer getExplicitLac() {
```

Comment:
sure we can discuss this in this week's meeting.

I think there should be only one LAC concept in the bookie server side. The LAC is indicating the last add confirmed for a ledger, no matter it is a piggyback lac or explicit lac. piggyback and explicit are just different ways on advancing the LAC. It should be unified. Otherwise, it is not going to scale.

You patch here is about advancing lac while our change is about notifications when lac is advanced. they are not conflicted and are actually resolving different problems. I would like to see a unified solution - e.g have only one lac or lac buffer in the FileInfo. it would be updated each time lac is advanced (no matter it is advanced or piggybacked) and notify any listeners listening on the lac changes.

Also, I don't like the approach that we did for digesting/checksumming the add entries, as it already made the server hard to verify digestion/checksum before persisting the entries. We should not use this approach for explicit lac. Since we are introducing new type of requests, we probably should use this chance to make things right.